tools/ci/actions/runner.sh

```diff
@@ -121,10 +121,23 @@ EOF
 
 }
 
+# ReportBuildStatus accepts an exit code as a parameter (defaults to 1) and also
+# instrucrts GitHub Actions to set build-status to 'failed' on non-zero exit or
+# 'success' otherwise.
+ReportBuildStatus () {
+  CODE=${1:-1}
+  if ((CODE)); then
+    STATUS='failed'
+  else
+    STATUS='success'
+  fi
+  echo "::set-output name=build-status::$STATUS"
+  exit $CODE
+}
+
 BasicCompiler () {
-  # The presence of this file can be detected by later steps which
-  # can choose to skip, rather than run (and presumably error).
-  touch compiler-failed-to-build
+  trap ReportBuildStatus ERR
```

Comment:
Well this magic is somehow more magical than the touch-a-file protocol (it requires more expert knowledge of `bash` to be understood), but it also makes the code cleaner.

Reply:
Indeed - having realised (while working on another workflow) that the file trick was not _necessary_, what bothers me is that we have other checks which check for the tree being clean, etc.